third-party/llvm/Makefile

```diff
@@ -36,6 +36,30 @@ LLVM_CONFIG_FILE := $(LLVM_INSTALL_DIR)/bin/llvm-config
 LLVM_CLANG_FILE := $(LLVM_INSTALL_DIR)/bin/clang
 LLVM_CLANG_CONFIG_FILE := $(LLVM_INSTALL_DIR)/configured-clang-sysroot-arguments
 
+# Sometimes one is using a custom version of GCC. In that event,
+# we'd like the built clang to use that version's std C++ headers, rather
+# than only finding the versions in /usr/include. To do that, pass
+# GCC_INSTALL_PREFIX for GCC builds.
+CHPL_LLVM_GCC_PREFIX := ""
+ifneq (, $(findstring gnu,$(CHPL_MAKE_HOST_COMPILER)))
+  CHPL_LLVM_GCC_PREFIX_DIR := $(shell ./find-gcc-prefix.sh $(CC))
+  ifneq ($(CHPL_LLVM_GCC_PREFIX_DIR),)
+    CHPL_LLVM_GCC_PREFIX := -DGCC_INSTALL_PREFIX=$(CHPL_LLVM_GCC_PREFIX_DIR)
+  endif
+endif
+
+# Decide if we want to dynamically link or not
+# This sets the default - it could be overridden during the make
+CHPL_LLVM_DYNAMIC := 1
+ifneq (, $(findstring cray,$(CHPL_MAKE_HOST_PLATFORM)))
```

Comment:
`cray` -> `cray-x` so we don't include `cray-cs`? Or do you want to do static builds for cray-cs as well?

Reply:
`ifneq (, $(filter cray-x%,$(CHPL_MAKE_HOST_PLATFORM)))` is the check we use cray-x* in other makefiles. Not sure if it pays to be consistent or not with that.